clients/adapter-kafka/src/main/java/org/eclipse/hono/adapter/client/command/kafka/KafkaBasedCommand.java

```diff
@@ -110,6 +109,45 @@ public static KafkaBasedCommand from(final KafkaConsumerRecord<String, Buffer> r
             throw new IllegalArgumentException("unsupported topic");
         }
         final String tenantId = honoTopic.getTenantId();
+        return from(record, tenantId);
+    }
+
+    /**
+     * Creates a command from a Kafka consumer record, forwarded by the Command Router.
+     * <p>
+     * The message is required to contain a <em>tenant_id</em> header and a <em>key</em> with the command
+     * target device id, matching the value of the <em>device_id</em> header.
+     * If that is not the case, an {@link IllegalArgumentException} is thrown.
+     * <p>
+     * In addition, the record is expected to contain
+     * <ul>
+     * <li>a <em>subject</em> header</li>
+     * <li>a non-empty <em>correlation-id</em> header if the <em>response-expected</em> header is set to {@code true}.</li>
+     * </ul>
+     * or otherwise the returned command's {@link #isValid()} method will return {@code false}.
+     * <p>
+     * If the <em>via</em> header is set, its value will be set as target gateway of the created command.
+     *
+     * @param record The record containing the command.
+     * @return The command.
+     * @throws NullPointerException if record is {@code null}.
+     * @throws IllegalArgumentException if the record doesn't correctly reference a target device.
+     */
+    public static KafkaBasedCommand fromRoutedCommandRecord(final KafkaConsumerRecord<String, Buffer> record) {
+        final String tenantId = getHeader(record, MessageHelper.APP_PROPERTY_TENANT_ID);
```

Comment:
Missing _null_ check for the `record` as described in the JavaDoc. If the `record` is null, then the `getHeader(...)` will throw a _null_ pointer exception.

Reply:
I've added the explicit null check.